### PR TITLE
add engine-role

### PR DESCRIPTION
**Pull request by hehechen**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: proxy_tests/proxy/config.rs [hunk 1]
```diff
@@ -172,6 +172,72 @@ fn test_config_proxy_engine_label() {
     );
 }
 
+#[test]
+fn test_config_proxy_engine_role_label() {
```
Comment (CalvinNeo) [inline proxy_tests/proxy/config.rs:176]: Is it a better option that we add this into `test_config_proxy_engine_label`. I think these tests have something in common.

**Review by breezewish — APPROVED** [review]

**Review by lidezhu — APPROVED** [review]

**Pull request merged**